src/vs-bicep/Bicep.VSLanguageServerClient.IntegrationTests/BicepConfigTests.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.IO;
+using System.Linq;
+using Bicep.VSLanguageServerClient.IntegrationTests.Utilities;
+using Microsoft.Test.Apex.VisualStudio.Editor;
+using Microsoft.Test.Apex.VisualStudio.Shell;
+using Microsoft.Test.Apex.VisualStudio.Solution;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Bicep.VSLanguageServerClient.IntegrationTests
+{
+    [TestClass]
+    public class BicepConfigTests : VisualStudioBicepHostTest
+    {
+        [TestMethod]
+        public void ChangeInLinterRuleLevelInBicepConfigFile_ShouldUpdateDiagnostics()
+        {
+            ProjectItemTestExtension projectItem = TestProject![@"Validation\main.bicep"];
+            IVisualStudioTextEditorTestExtension editor = projectItem.GetDocumentAsTextEditor().Editor;
+
+            WaitForBicepLanguageServiceActivation(editor);
+
+            var errorListService = VsHostUtility.VsHost!.ObjectModel.Shell.ToolWindows.ErrorList;
+            ErrorItemTestExtension[] diagnostics = errorListService.AllItems;
+
+            Assert.IsNotNull(diagnostics);
+            Assert.AreEqual(1, diagnostics.Length);
+
+            Assert.IsTrue(diagnostics.Any(
+                d => d.Description.Contains("Parameter \"cosmosDBDatabaseName\" is declared but never used.") &&
+                d.ErrorLevel == ErrorListErrorLevel.Warning &&
+                d.LineNumber == 1 &&
+                d.Column == 7));
+
+            CreateBicepConfigFile(projectItem.FullPath, @"{
+  ""analyzers"": {
+    ""core"": {
+      ""verbose"": false,
+      ""enabled"": true,
+      ""rules"": {
+        ""no-unused-params"": {
+          ""level"": ""error""
+        }
+      }
+    }
+  }
+}");
+            // Re trigger diagnostics
```

Comment:
Do we not need some way of waiting until the diagnostics change?

Reply:
Good idea.

Updated here:
https://github.com/Azure/bicep/commit/baf901c0cf6153047bdd1d2b037f22920359eae4